tests/test_models.py

```diff
@@ -0,0 +1,172 @@
+        self.assertEqual(cache.get(key), cm3)
+
+    def test_queries_count_after_saving_and_filter(self):
+        with self.assertNumQueries(1):
+            self.test_instance_is_cached_after_saving()
+            cm = models.CacheableMock.objects.filter(pk=1)
+            self.assertEqual(len(cm), 1)
+
+    def test_queries_count_after_saving_and_get(self):
+        with self.assertNumQueries(1):
+            self.test_instance_is_cached_after_saving()
+            cm = models.CacheableMock.objects.get(pk=1)
+            self.assertEqual(cm.pk, 1)
+
+    def test_queries_count_after_create_and_filter(self):
+        with self.assertNumQueries(1):
+            self.test_instance_is_cached_after_create()
+            cm = models.CacheableMock.objects.filter(pk=1)
+            self.assertEqual(len(cm), 1)
+
+    def test_queries_count_after_create_and_get(self):
+        with self.assertNumQueries(1):
+            self.test_instance_is_cached_after_create()
+            cm = models.CacheableMock.objects.get(pk=1)
+            self.assertEqual(cm.pk, 1)
+
+    def test_queries_count_after_get_or_create_and_filter(self):
+        """ 4 queries here, because of transaction SAVEPOINT """
+        with self.assertNumQueries(4):
+            self.test_instance_is_cached_after_get_or_create()
+            cm = models.CacheableMock.objects.filter(pk=1)
+            self.assertEqual(len(cm), 1)
+
+    def test_queries_count_after_get_or_create_and_get(self):
+        """ 4 queries here, because of transaction SAVEPOINT """
+        with self.assertNumQueries(4):
+            self.test_instance_is_cached_after_get_or_create()
+            cm = models.CacheableMock.objects.get(pk=1)
+            self.assertEqual(cm.pk, 1)
+
+    def test_multiple_objects_returned_with_filters(self):
+        with self.assertNumQueries(3):
+            models.CacheableMock.objects.create()
+            models.CacheableMock.objects.create()
+            models.CacheableMock.objects.create()
+            cms = models.CacheableMock.objects.filter(pk__in=[1, 3])
+            self.assertEqual(len(cms), 2)
+            cm = models.CacheableMock.objects.filter(pk=1)
+            self.assertEqual(len(cm), 1)
+
+    def test_cached_instance_after_update(self):
+        models.CacheableMock.objects.create()
+        key = get_key_for_instance(models.CacheableMock.objects.get(pk=1))
+        cms = models.CacheableMock.objects.filter(pk=1)
+        cms.update(name='test')
+        for cm in cms:
+            cm.save()
+        cm = models.CacheableMock.objects.get(pk=1)
+        self.assertEqual(cache.get(key), cm)
+        self.assertEqual(cm.name, 'test')
+
+
+class SlugableModelTests(TestCase):
+
+    def setUp(self):
+        self.sm = models.SlugableMock.objects.create(
+            label='Never play leapfrog on a unicorn!')
+
+    def test_simple_sluggable_model_creation(self):
+        self.assertEquals(self.sm.slug, 'never-play-leapfrog-on-a-unicorn')
```

Comment:
Hey @harmo, I believe that the slug should have been `never-play-leapfrog-on-a-unicorn` for the object `SlugableMock.objects.create(label='Never play leapfrog on a unicorn!')`. Instead we have: `slugablemock-object`.

What do you think?